Geometry_Engine/Modify/RemoveLeastSignificantVertices.cs

```diff
@@ -67,8 +61,18 @@ public static Polyline RemoveLeastSignificantVertices(this Polyline polyline, do
                     startIndex++; //Move onto the next point
             }
 
-            if (pnts.First() != pnts.Last())
-                pnts.Add(pnts.First()); //Reclose polyline
+            if (!polyline.IsClosed())
+            {
+                if (pnts.Last() != originalPnts.Last())
```

Comment:
This check is dangerous - the != operator does not take any tolerance into account, can imagine this being an issue. I would rather skip this line at all and replace the last point with the original last point `if (!polyline.IsClosed())`